containers/R/rclipper_user/vignettes/Rclipper.Rmd

```diff
@@ -0,0 +1,133 @@
+[clipper_admin package](https://pypi.python.org/pypi/clipper_admin).
+
+## Deploying a model
+Once you've built a model, use the provided command to deploy it with the 
+[clipper_admin package](https://pypi.python.org/pypi/clipper_admin). For information about how to
+register the model with an application so that it can be queried, please consult the 
+[clipper_admin API documentation](http://docs.clipper.ai/en/).
+
+## Querying a model
+After you've built a model, deployed the model, and registered the model with an application,
+you can query it with the input data of the correct type. The following table maps the input type of 
+your model's prediction function to the Clipper input type associated with your deployed model:
+
+| R input type   | Clipper Input Type |      JSON Format      | Example                        |
+|---------------:|:-------------------|:---------------------:|:------------------------------:|
+| Raw Vector     |   Bytes            | Base64-encoded string | Y2xpcHBlciB0ZXh0               |
+| Integer Vector |   Ints             | Integer array         | [1,2,3,4]                      |
+| Numeric Vector |   Floats           | Floating point array  | [1.0,2.0,3.0.,4.0]             | 
+| String         |   Strings          | String                | "input text"                   |
+| Data Frame     |   Strings          | String                | `jsonlite::toJSON(mtcars)`     |
+| Matrix         |   Strings          | String                | `jsonlite::toJSON(diag(3))`    |
+| Array          |   Strings          | String                | `jsonlite::toJSON(array(1:4))` |
+| List           |   Strings          | String                | `jsonlite::toJSON(list(1:4))`  |
+
+## Example
+
+### Define an API-compatible prediction function
+```{r}
+#' Given a list of vector inputs,
+#' outputs a list containing the
+#' length of each input vector as a string
+pred_fn = function(inputs) {
+  return(lapply(inputs, function(input) {
+    return(as.character(length(input)))
+  }))
+}
+
+print(pred_fn(list(c(1,2), c(3))))
+```
+
+### Build a model
+```{r}
+# Specify that the prediction function expects integer vectors
+# by supplying an integer vector as the sample input
+Rclipper::build_model("test-model", "1", pred_fn, sample_input = as.integer(c(1,2,3)))
+```
+
+### Deploy and link the model
+This assumes that a Clipper cluster is running on *localhost* with a registered application that has the name *app1*.
+In a Python interactive environment:
+```
+from clipper_admin import DockerContainerManager, ClipperConnection
+cm = DockerContainerManager()
+conn = ClipperConnection(cm)
+conn.connect()
+
+# Deploy a single replica of the model
+conn.deploy_model("test-model", "1", "ints", "test-model:1", replicas=1)
+
+conn.link_model_to_app("app1", "test-model")
+```
+
+### Query the model
+We can now query the model via the provided application with [cURL](https://github.com/curl/curl):
```

Comment:
"You can now query the model from any HTTP client. For example, directly from the command line with [cURL]..."